### PR TITLE
Add one trust libraries

### PR DESCRIPTION
**Pull request by juanmanzojr**
This PR introduces changes to self-host the OneTrust libraries instead of relying on external sources.

The changes for self hosting OneTrust are behind the `'onetrust_cookie_scripts'` DCDO flag.

Changes made in to: `shared/haml/onetrust_cookie_scripts.haml`.

## Links

- jira ticket: [here](https://codedotorg.atlassian.net/browse/P20-779)

## Testing story
Working on writing tests in another PR

### CODE REVIEW

**Review by snickell — APPROVED** [review]
LGTM from a git lfs perspective

File: shared/haml/onetrust_cookie_scripts.haml [hunk 1]
```diff
@@ -19,8 +21,11 @@
 -# domain was given.
 - one_trust_domain = one_trust_domains.dig(domain, cookie_script_env)
 - if cookie_script_env != 'off' && one_trust_domain.present?
-  - auto_block_src = "https://cdn.cookielaw.org/consent/#{one_trust_domain}/OtAutoBlock.js"
-  %script{src: auto_block_src, type: 'text/javascript'}
-  %script{src: 'https://cdn.cookielaw.org/scripttemplates/otSDKStub.js', type: 'text/javascript', charset: 'UTF-8', 'data-domain-script' => one_trust_domain}
+  - base_url = onetrust_version == 'self_hosted' ? "/onetrust" : "https://cdn.cookielaw.org"
```
Comment (daynew) [inline shared/haml/onetrust_cookie_scripts.haml:24]: Does this work on hourofcode.com or code.org? During our testing, I thought the url had to be studio.code.org/onetrust/

**Review by daynew — CHANGES_REQUESTED** [review]

File: shared/haml/onetrust_cookie_scripts.haml (hunk 1 above)
Comment (juanmanzojr) [reply shared/haml/onetrust_cookie_scripts.haml:24]: This works for hourofcode.com
DCDO flag set to 'self-hosted'. You can see one trust in the sources
Screenshot 2024-08-26 at 3 09 47 PM

Will check for studio.code.org

Comment (juanmanzojr) [reply shared/haml/onetrust_cookie_scripts.haml:24]: Screenshot 2024-08-26 at 4 13 10 PM
Heres studio.code.org

**Review by daynew — APPROVED** [review]

**Pull request merged**